aws/resource_aws_iam_role.go

```diff
@@ -400,3 +402,15 @@ func resourceAwsIamRoleDelete(d *schema.ResourceData, meta interface{}) error {
 		return nil
 	})
 }
+
+func getIAMRoleNameFromIAMRoleARN(inputARN string) (string, error) {
```

Comment:
When passed an ARN of a different resource type, this function will still return the contents of `parsedARN.Resource` unmodified (because of the way `TrimPrefix` works).

Since this function's name hints at being specific to roles, would it make sense to err when the resource is not a role?

Better yet, might it make sense to call it something generic, like maybe `getResourceNameFromARN`?

Reply:
If it's generic, it'd probably have to handle the other prefixes: "user/", "group/", "policy/", "instance-profile/" etc. of which there are many. It might be safe to remove everything up through the first slash, but it's hard to be sure if that's right.

Reply:
@rehevkor5 That's exactly how I was imagining it working.
Only thing is to make sure that there are no rogue resources not adhering to `type/name` format for that part of the ARN.

Reply:
I was afraid that would be the case when I mentioned rogue resources.
I remember running into those unusual ARNs every now and then.

Oh well, never mind then. Not worth the trouble.